src/access_nri_intake/cli.py

```diff
@@ -87,18 +94,19 @@ def _check_build_args(args_list):
 
     if len(names) != len(set(names)):
         seen = set()
-        dupes = [name for name in names if name in seen or seen.add(name)]
+        dupes = [name for name in names if name in seen or seen.add(name)]  # type: ignore
+        # seen.add(name) returns None & so is always Falsey - so what is it doing?
```

Comment:
Please tell me `Falsey` is a British programming idiom :P

Reply:
Seriously though, I think the `or` is being used as a sneaky way to avoid an `if else` statement to add things to the "I've now seen this name" set (`seen`).

It's going in order through the `names` list. The `or` is evaluated left to right, so:

- If the name has been seen earlier in the list (i.e., `name in seen == True`), then it gets added to `dupes`. The `seen.add(name)` is never reached.
- If `name in seen == False`, then it moves on to the second part of the `or` statement, which adds `name` to the `seen` set. You're right that it's always False, but that's the point - it's `False` on _this_ pass, but now that it's been added to the `seen` set, if the same name recurs on a future pass, it is now known to be a duplicate.

(Interestingly, `False or None` evaluates to `None`, not `False`, but for this list comprehension I'm assuming anything not `True` is close enough to `False` to not care.)

```ipython
In [13]: (False or None) is None
Out[13]: True

In [14]: None or False
Out[14]: False
```

Reply:
Also, if a duplicated is repeated `N` times, then it will appear `(N-1)` times in the error message below - not sure if that's a bug or feature.

Reply:
Funnily enough, I think `Falsey` is pretty common worldwide - just googled it and it looks like it mostly crops up in javascript land though.

More importantly though, I had completely forgotten that the `or` will get executed left -> right & that `seen.add()` will still evaluate, even though it returns None.

I think the `False or None == None` and `None or False == False` thing is the result of the same trick - `None` and `False` are both falsey, so we wind up getting the second value in either case. Makes perfect sense, but completely counterintuitive.

Either way, I'll remove the unnecessary comment & see if I can strip it down to 1 error message, not N-1.